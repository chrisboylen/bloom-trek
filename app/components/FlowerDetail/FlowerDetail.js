import React, { Component } from 'react';
import {
  StyleSheet, Text, Image, View, ScrollView,
} from 'react-native';
import GestureRecognizer from 'react-native-swipe-gestures';
import PropTypes from 'prop-types';

export default class FlowerDetail extends Component {
  onSwipeRight() {
    const { resetSelectedFlower } = this.props;
    resetSelectedFlower();
  }

  render() {
    const { flowerInfo } = this.props;
    const {
      flower_img_url,
      name,
      scientific_name,
      description,
      habitat,
    } = flowerInfo;

    return (
      <GestureRecognizer style={styles.container} onSwipeRight={state => this.onSwipeRight(state)}>
        <View style={styles.imageContainer}>
          <Image
            style={styles.image}
            source={{ uri: flower_img_url }}
          />
        </View>
        <ScrollView style={styles.text}>
          <Text style={styles.info}><Text style={styles.emphasis}>Name: </Text>{name}</Text>
          <Text style={styles.info}><Text style={styles.emphasis}>Scientific Name: </Text>{scientific_name}</Text>
          <Text style={styles.info}><Text style={styles.emphasis}>Habitat: </Text>{habitat}</Text>
          <Text style={styles.info}><Text style={styles.emphasis}>Description: </Text>{description}</Text>
        </ScrollView>
      </GestureRecognizer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: '85%',
  },
  emphasis: {
    color: '#b55760',
    fontSize: 15,
    fontWeight: 'bold',
  },
  info: {
    marginBottom: 5,
  },
  text: {
    height: '35%',
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 5,
    marginBottom: 10,
  },
  imageContainer: {
    height: '50%',
    width: '100%',
  },
  image: {
    height: '100%',
  },
});

FlowerDetail.propTypes = {
  resetSelectedFlower: PropTypes.func,
  flowerInfo: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
};
