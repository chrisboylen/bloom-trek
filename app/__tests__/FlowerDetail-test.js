import React from 'react';
import { shallow } from 'enzyme';
import FlowerDetail from '../components/FlowerDetail/FlowerDetail';

describe('FlowerDetail', () => {
  let mockResetSelectedFlower;
  let wrapper;
  let mockFlowerInfo;

  beforeEach(() => {
    mockResetSelectedFlower = jest.fn();
    mockFlowerInfo = {
      id: 71,
      flower_img_url: 'http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg',
      name: 'Butter And Eggs',
      common_name: 'Yellow Toadflax',
      scientific_name: 'Linaria vulgaris',
      description: 'Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).',
      bloom_start: 6,
      bloom_end: 10,
      habitat: 'Open areas around former homesteads.',
    };
    wrapper = shallow(
      <FlowerDetail
        flowerInfo={mockFlowerInfo}
        resetSelectedFlower={mockResetSelectedFlower}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('onSwipeRight', () => {
    it('should invoke resetSelectedFlower when invoked', () => {
      wrapper.instance().onSwipeRight();

      expect(mockResetSelectedFlower).toHaveBeenCalled();
    });
  });
});
