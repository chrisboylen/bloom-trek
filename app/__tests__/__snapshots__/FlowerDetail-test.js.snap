// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowerDetail should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlowerDetail
    flowerInfo={
      Object {
        "bloom_end": 10,
        "bloom_start": 6,
        "common_name": "Yellow Toadflax",
        "description": "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
        "flower_img_url": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
        "habitat": "Open areas around former homesteads.",
        "id": 71,
        "name": "Butter And Eggs",
        "scientific_name": "Linaria vulgaris",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "height": "50%",
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </Component>,
        <ScrollViewMock
          style={
            Object {
              "height": "35%",
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#005900",
                  "fontSize": 15,
                  "fontWeight": "bold",
                }
              }
            >
              Name: 
            </Text>
            BUTTER AND EGGS
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#005900",
                  "fontSize": 15,
                  "fontWeight": "bold",
                }
              }
            >
              Scientific Name: 
            </Text>
            Linaria vulgaris
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#005900",
                  "fontSize": 15,
                  "fontWeight": "bold",
                }
              }
            >
              Habitat: 
            </Text>
            Open areas around former homesteads.
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#005900",
                  "fontSize": 15,
                  "fontWeight": "bold",
                }
              }
            >
              Description: 
            </Text>
            Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).
          </Text>
        </ScrollViewMock>,
      ],
      "onSwipeRight": [Function],
      "style": Object {
        "height": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            source={
              Object {
                "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
              }
            }
          />,
          "style": Object {
            "height": "50%",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
            },
            "style": Object {
              "height": "100%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Name: 
              </Text>
              BUTTER AND EGGS
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Scientific Name: 
              </Text>
              Linaria vulgaris
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Habitat: 
              </Text>
              Open areas around former homesteads.
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Description: 
              </Text>
              Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).
            </Text>,
          ],
          "style": Object {
            "height": "35%",
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Name: 
                </Text>,
                "BUTTER AND EGGS",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "marginBottom": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Name: ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": "Name: ",
                "type": [Function],
              },
              "BUTTER AND EGGS",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Scientific Name: 
                </Text>,
                "Linaria vulgaris",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "marginBottom": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Scientific Name: ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": "Scientific Name: ",
                "type": [Function],
              },
              "Linaria vulgaris",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Habitat: 
                </Text>,
                "Open areas around former homesteads.",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "marginBottom": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Habitat: ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": "Habitat: ",
                "type": [Function],
              },
              "Open areas around former homesteads.",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Description: 
                </Text>,
                "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "marginBottom": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Description: ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": "Description: ",
                "type": [Function],
              },
              "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
            ],
            "type": [Function],
          },
        ],
        "type": [MockFunction],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "height": "50%",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                }
              }
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </Component>,
          <ScrollViewMock
            style={
              Object {
                "height": "35%",
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Name: 
              </Text>
              BUTTER AND EGGS
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Scientific Name: 
              </Text>
              Linaria vulgaris
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Habitat: 
              </Text>
              Open areas around former homesteads.
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#005900",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
              >
                Description: 
              </Text>
              Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).
            </Text>
          </ScrollViewMock>,
        ],
        "onSwipeRight": [Function],
        "style": Object {
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={
                Object {
                  "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
                }
              }
              style={
                Object {
                  "height": "100%",
                }
              }
            />,
            "style": Object {
              "height": "50%",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "uri": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
              },
              "style": Object {
                "height": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginBottom": 5,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Name: 
                </Text>
                BUTTER AND EGGS
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginBottom": 5,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Scientific Name: 
                </Text>
                Linaria vulgaris
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginBottom": 5,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Habitat: 
                </Text>
                Open areas around former homesteads.
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginBottom": 5,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Description: 
                </Text>
                Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).
              </Text>,
            ],
            "style": Object {
              "height": "35%",
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#005900",
                        "fontSize": 15,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Name: 
                  </Text>,
                  "BUTTER AND EGGS",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "marginBottom": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Name: ",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    },
                  },
                  "ref": null,
                  "rendered": "Name: ",
                  "type": [Function],
                },
                "BUTTER AND EGGS",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#005900",
                        "fontSize": 15,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Scientific Name: 
                  </Text>,
                  "Linaria vulgaris",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "marginBottom": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Scientific Name: ",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    },
                  },
                  "ref": null,
                  "rendered": "Scientific Name: ",
                  "type": [Function],
                },
                "Linaria vulgaris",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#005900",
                        "fontSize": 15,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Habitat: 
                  </Text>,
                  "Open areas around former homesteads.",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "marginBottom": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Habitat: ",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    },
                  },
                  "ref": null,
                  "rendered": "Habitat: ",
                  "type": [Function],
                },
                "Open areas around former homesteads.",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#005900",
                        "fontSize": 15,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Description: 
                  </Text>,
                  "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "marginBottom": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Description: ",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#005900",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    },
                  },
                  "ref": null,
                  "rendered": "Description: ",
                  "type": [Function],
                },
                "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
              ],
              "type": [Function],
            },
          ],
          "type": [MockFunction],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
